diff --git a/src/main/java/de/ddm/actors/Guardian.java b/src/main/java/de/ddm/actors/Guardian.java
index 2a3d378..6ba6db7 100644
--- a/src/main/java/de/ddm/actors/Guardian.java
+++ b/src/main/java/de/ddm/actors/Guardian.java
@@ -68,3 +68,3 @@ public class Guardian extends AbstractBehavior<Guardian.Message> {

-		this.reaper = context.spawn(Reaper.create(), Reaper.DEFAULT_NAME);
+		de.ddm.singletons.ReaperSingleton.set(context.spawn(Reaper.create(), Reaper.DEFAULT_NAME));
 		this.master = this.isMaster() ? context.spawn(Master.create(), Master.DEFAULT_NAME) : null;
@@ -90,3 +90,2 @@ public class Guardian extends AbstractBehavior<Guardian.Message> {

-	private final ActorRef<Reaper.Message> reaper;
 	private ActorRef<Master.Message> master;
diff --git a/src/main/resources/application.conf b/src/main/resources/application.conf
index fb1072e..7710f3a 100644
--- a/src/main/resources/application.conf
+++ b/src/main/resources/application.conf
@@ -104,2 +104,3 @@ akka {
   logging-filter = "akka.event.slf4j.Slf4jLoggingFilter"
+  logger-startup-timeout = "60s"

