diff --git a/pom.xml b/pom.xml
index c9c493c..471b3e8 100644
--- a/pom.xml
+++ b/pom.xml
@@ -127,4 +127,5 @@
 						</goals>
 						<configuration>
+							<finalName>app</finalName>
 							<transformers>
 								<transformer
diff --git a/src/main/java/de/ddm/Main.java b/src/main/java/de/ddm/Main.java
index 5592a53..f4138ad 100644
--- a/src/main/java/de/ddm/Main.java
+++ b/src/main/java/de/ddm/Main.java
@@ -24,7 +24,4 @@ public class Main {
 			guardian.tell(new Guardian.StartMessage());
 
-			waitForInput(">>> Press ENTER to exit <<<");
-
-			guardian.tell(new Guardian.ShutdownMessage());
 		}
 	}
diff --git a/src/main/java/de/ddm/actors/Guardian.java b/src/main/java/de/ddm/actors/Guardian.java
index 2a3d378..263b88d 100644
--- a/src/main/java/de/ddm/actors/Guardian.java
+++ b/src/main/java/de/ddm/actors/Guardian.java
@@ -68,4 +68,5 @@ public class Guardian extends AbstractBehavior<Guardian.Message> {
 
 		this.reaper = context.spawn(Reaper.create(), Reaper.DEFAULT_NAME);
+		de.ddm.singletons.ReaperSingleton.set(this.reaper);
 		this.master = this.isMaster() ? context.spawn(Master.create(), Master.DEFAULT_NAME) : null;
 		this.worker = context.spawn(Worker.create(), Worker.DEFAULT_NAME);
diff --git a/src/main/java/de/ddm/actors/Master.java b/src/main/java/de/ddm/actors/Master.java
index 8fa9176..666a4ef 100644
--- a/src/main/java/de/ddm/actors/Master.java
+++ b/src/main/java/de/ddm/actors/Master.java
@@ -44,4 +44,17 @@ public class Master extends AbstractBehavior<Master.Message> {
 	private Master(ActorContext<Message> context) {
 		super(context);
+		try {
+			if(context.getSelf() == null){
+				System.out.println("Context self is null, sleeping for a second to workaround lazy initialization");
+				Thread.sleep(1000); // Sleep for a second to workaround context.getSelf() returning null
+			} else if (de.ddm.singletons.ReaperSingleton.get() == null) {
+				System.out.println("Reaper is null, sleeping for a second to workaround lazy initialization");
+				Thread.sleep(1000); // Sleep for a second to workaround ReaperSingleton.get() returning null
+			} else {
+				System.out.println("Worker actor created with path: " + context.getSelf().path().toString());
+			}
+		} catch (InterruptedException e) {
+			throw new RuntimeException(e);
+		}
 		Reaper.watchWithDefaultReaper(this.getContext().getSelf());
 
diff --git a/src/main/java/de/ddm/actors/Worker.java b/src/main/java/de/ddm/actors/Worker.java
index a3584c8..66cf8f1 100644
--- a/src/main/java/de/ddm/actors/Worker.java
+++ b/src/main/java/de/ddm/actors/Worker.java
@@ -43,4 +43,17 @@ public class Worker extends AbstractBehavior<Worker.Message> {
 	private Worker(ActorContext<Message> context) {
 		super(context);
+		try {
+			if(context.getSelf() == null){
+				System.out.println("Context self is null, sleeping for a second to workaround lazy initialization");
+				Thread.sleep(1000); // Sleep for a second to workaround context.getSelf() returning null
+			} else if (de.ddm.singletons.ReaperSingleton.get() == null) {
+				System.out.println("Reaper is null, sleeping for a second to workaround lazy initialization");
+				Thread.sleep(1000); // Sleep for a second to workaround ReaperSingleton.get() returning null
+			} else {
+				System.out.println("Worker actor created with path: " + context.getSelf().path().toString());
+			}
+		} catch (InterruptedException e) {
+			throw new RuntimeException(e);
+		}
 		Reaper.watchWithDefaultReaper(this.getContext().getSelf());
 
diff --git a/src/main/java/de/ddm/configuration/Command.java b/src/main/java/de/ddm/configuration/Command.java
index 61585ef..18f9deb 100644
--- a/src/main/java/de/ddm/configuration/Command.java
+++ b/src/main/java/de/ddm/configuration/Command.java
@@ -14,7 +14,13 @@ public abstract class Command {
 	String host = SystemConfigurationSingleton.get().getHost();
 
+	@Parameter(names = {"-ia", "--ipAddress"}, description = "This machine's IP that we use to bind this application against", required = false)
+	String ipAddress = SystemConfigurationSingleton.get().getIpAddress();
+
 	@Parameter(names = {"-p", "--port"}, description = "This machines port that we use to bind this application against", required = false)
 	int port = this.getDefaultPort();
 
+	@Parameter(names = {"-kb", "--runningInKubernetes"}, description = "The application is running in Kubernetes", required = false, arity = 1)
+	boolean runningInKubernetes = SystemConfigurationSingleton.get().isRunningInKubernetes();
+
 	@Parameter(names = {"-w", "--numWorkers"}, description = "The number of workers (indexers/validators) to start locally; should be at least one if the algorithm is started standalone (otherwise there are no workers to run the discovery)", required = false)
 	int numWorkers = SystemConfigurationSingleton.get().getNumWorkers();
diff --git a/src/main/java/de/ddm/configuration/SystemConfiguration.java b/src/main/java/de/ddm/configuration/SystemConfiguration.java
index c41e726..b324a2f 100644
--- a/src/main/java/de/ddm/configuration/SystemConfiguration.java
+++ b/src/main/java/de/ddm/configuration/SystemConfiguration.java
@@ -25,4 +25,6 @@ public class SystemConfiguration {
 	private int masterPort = DEFAULT_MASTER_PORT;      // The port of the master; if this is a master, masterPort = port
 
+	private String ipAddress = getDefaultHost();            // The IP address of this machine
+
 	private String actorSystemName = "ddm";            // The name of this application
 
@@ -33,4 +35,6 @@ public class SystemConfiguration {
 	private boolean hardMode = false;					// Solve the hard version of the task
 
+	private boolean runningInKubernetes = false;        // The application is running in Kubernetes
+
 	private static String getDefaultHost() {
 		try {
@@ -45,4 +49,5 @@ public class SystemConfiguration {
 		this.host = commandMaster.host;
 		this.port = commandMaster.port;
+		this.ipAddress = commandMaster.ipAddress;
 		this.masterHost = commandMaster.host;
 		this.masterPort = commandMaster.port;
@@ -50,4 +55,5 @@ public class SystemConfiguration {
 		this.startPaused = commandMaster.startPaused;
 		this.hardMode = commandMaster.hardMode;
+		this.runningInKubernetes = commandMaster.runningInKubernetes;
 	}
 
@@ -56,15 +62,21 @@ public class SystemConfiguration {
 		this.host = commandWorker.host;
 		this.port = commandWorker.port;
+		this.ipAddress = commandWorker.ipAddress;
 		this.masterHost = commandWorker.masterhost;
 		this.masterPort = commandWorker.masterport;
 		this.numWorkers = commandWorker.numWorkers;
+		this.runningInKubernetes = commandWorker.runningInKubernetes;
 	}
 
 	public Config toAkkaConfig() {
 		return ConfigFactory.parseString("" +
-				"akka.remote.artery.canonical.hostname = \"" + this.host + "\"\n" +
-				"akka.remote.artery.canonical.port = " + this.port + "\n" +
-				"akka.cluster.roles = [" + this.role + "]\n" +
-				"akka.cluster.seed-nodes = [\"akka://" + this.actorSystemName + "@" + this.masterHost + ":" + this.masterPort + "\"]")
+						"akka.remote.artery.canonical.hostname = \"" + this.host + "\"\n" +
+						"akka.remote.artery.canonical.port = " + this.port + "\n" +
+						(this.runningInKubernetes ?
+								"akka.remote.artery.bind.hostname = \"" + this.ipAddress + "\"\n" +
+										"akka.remote.artery.bind.port = " + this.port + "\n" : "" ) +
+						"akka.logger-startup-timeout = 60s\n" +
+						"akka.cluster.roles = [" + this.role + "]\n" +
+						"akka.cluster.seed-nodes = [\"akka://" + this.actorSystemName + "@" + this.masterHost + ":" + this.masterPort + "\"]")
 				.withFallback(ConfigFactory.load("application"));
 	}
